core/trino-main/src/main/java/io/trino/execution/resourcegroups/InternalResourceGroup.java

```diff
@@ -785,11 +820,11 @@ private void queryFinished(ManagedQueryExecution query)
             // The query is present in runningQueries
             if (lastUsage != null) {
                 // CPU is measured cumulatively (i.e. total CPU used until this moment by the query). Memory is measured
-                // instantaneously (how much memory the query is using at this moment). At query completion, memory usage
-                // drops to zero.
+                // instantaneously (how much memory the query is using at this moment). At query completion, memory and
+                // data scan usage drops to zero.
                 ResourceUsage finalUsage = new ResourceUsage(
                         query.getTotalCpuTime().toMillis(),
-                        0L);
+                        0L, 0L);
```

Comment:
I don't think resetting the `physicalInputDataSize` to zero here is how this should work. The `physicalInputDataSize` is more like CPU and not memory in the sense that once the bytes are read, they do not occupy the cluster. If you want to punish "bad" users who use too much network, you should make this limit period-based and add regeneration similar to `io.trino.execution.resourcegroups.InternalResourceGroup#internalGenerateCpuQuota`

Reply:
I would add a new property because the "units" are different (time vs bytes)

Reply:
@lukasz-stec I have made the changes to support period-based tracking by adding a new property `physicalDataScanQuotaPeriod`. Please take a look, thanks.